pkg/controller/dataimportcron-controller.go

```diff
@@ -311,13 +314,40 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	dataImportCronCopy := dataImportCron.DeepCopy()
 	imports := dataImportCron.Status.CurrentImports
 	importSucceeded := false
+
+	dataVolume := dataImportCron.Spec.Template
+	explicitScName := getStorageClassFromTemplate(&dataVolume)
+	dvStorageClass, err := cc.GetStorageClassByName(r.client, explicitScName)
+	if err != nil {
+		return res, err
+	}
+	format, err := r.getSourceFormat(ctx, dataImportCron, dvStorageClass)
+	if err != nil {
+		return res, err
+	}
+	snapshot, err := r.getSnapshot(ctx, dataImportCron, format)
+	if err != nil {
+		return res, err
+	}
+
+	handleDataVolumePopulated := func() error {
+		if err = r.updateSourceObject(ctx, dataImportCron, pvc); err != nil {
```

Comment:
You mix `err = ` and `err := ` in several ifs here. I think the later is better.

Reply:
yep, changed to `err :=` throughout the controller